dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/deduplication/DeduplicationServiceIntegrationTest.java

```diff
@@ -71,24 +73,23 @@ public void setUpTest()
     void testGetAllPotentialDuplicateByDifferentStatus()
         throws PotentialDuplicateConflictException
     {
-        assertEquals( 0, deduplicationService.getAllPotentialDuplicates().size() );
         PotentialDuplicate potentialDuplicate = new PotentialDuplicate( teiA, teiB );
         deduplicationService.addPotentialDuplicate( potentialDuplicate );
         PotentialDuplicate potentialDuplicate1 = new PotentialDuplicate( teiC, teiD );
         deduplicationService.addPotentialDuplicate( potentialDuplicate1 );
-        PotentialDuplicateQuery potentialDuplicateQuery = new PotentialDuplicateQuery();
-        potentialDuplicateQuery.setTeis( Arrays.asList( teiA, teiC ) );
-        assertEquals( 2, deduplicationService.getAllPotentialDuplicatesBy( potentialDuplicateQuery ).size() );
+        PotentialDuplicateCriteria criteria = new PotentialDuplicateCriteria();
+        criteria.setTeis( Arrays.asList( teiA, teiC ) );
+        assertEquals( 2, deduplicationService.getPotentialDuplicates( criteria ).size() );
         // set one potential duplicate to invalid
         potentialDuplicate.setStatus( DeduplicationStatus.INVALID );
         deduplicationService.updatePotentialDuplicate( potentialDuplicate );
-        assertEquals( 2, deduplicationService.getAllPotentialDuplicates().size() );
-        potentialDuplicateQuery.setStatus( DeduplicationStatus.OPEN );
-        assertEquals( 1, deduplicationService.getAllPotentialDuplicatesBy( potentialDuplicateQuery ).size() );
-        potentialDuplicateQuery.setStatus( DeduplicationStatus.INVALID );
-        assertEquals( 1, deduplicationService.getAllPotentialDuplicatesBy( potentialDuplicateQuery ).size() );
-        potentialDuplicateQuery.setStatus( DeduplicationStatus.ALL );
-        assertEquals( 2, deduplicationService.getAllPotentialDuplicatesBy( potentialDuplicateQuery ).size() );
+
+        criteria.setStatus( DeduplicationStatus.OPEN );
+        assertEquals( 1, deduplicationService.getPotentialDuplicates( criteria ).size() );
```

Comment:
Such assertions especially in integration tests can be frustrating once the test is flaky and only fails on CI. They just tell us `expected 1, got 3` if the sizes don't match. Ideally we also see whats in the collection if sizes don't match.

What do you think of adding and using

```java
    /**
     * Asserts that the given collection is of the expected size. Use {@link #assertIsEmpty(Collection)} instead of asserting on size 0.
     *
     * @param expectedSize the expected size of actual
     * @param actual the actual collection to check
     */
    public static <T> void assertSize(int expectedSize, Collection<T> actual) {
        assertNotEmpty(actual);
        assertEquals(expectedSize, actual.size(), () -> String.format("Expected collection to contain %d elements, got %d instead: %s", expectedSize, actual.size(), actual));
    }
```

here https://github.com/dhis2/dhis2-core/blob/ecf7d9990ec39aaeded74cd287633022c7769493/dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/utils/Assertions.java#L112

gives us a message like

```
org.opentest4j.AssertionFailedError: Expected collection to contain 6 elements, got 3 instead: [1, 2, 3] ==> 
Expected :6
Actual   :3
```